packages/notebook/src/actions.tsx

```diff
@@ -309,9 +316,13 @@ namespace NotebookActions {
     let state = Private.getState(widget);
     let cells = widget.model.cells;
     let widgets = widget.widgets;
+    let focus = -1;
     cells.beginCompoundOperation();
     for (let i = 1; i < cells.length; i++) {
       if (widget.isSelectedOrActive(widgets[i])) {
+        if (focus == -1) {
+            focus = Math.max(0, i - 2);
```

Comment:
I am a little puzzled by this logic: why overshoot by two, rather than one? Also, when I experiment with this, it looks like it just scrolls the active cell into view, rather than overshooting, so I feel I am missing something here...

Reply:
Admittedly, I was just hacking this until it worked for me. I _believe_ what's going on is that later there's the line:
```
widget.select(widgets[i - 1]);
```
Then I want to focus the cell _before_ that so I set `focus = i - 2`. I would agree the implementation is a little ugly as it stands and hard to follow.